### ISSUE TITLE
Obsoleted update shouldn't be pushable back to testing

### ISSUE THREAD

**Issue by mattiaverga** [fedora-infra/bodhi#3976]
An update gets obsoleted when another one with a newer build is submitted.
I've just realized that obsoleted updates can be pushed back to testing by users... is that something we really want?

**Comment by AdamWill** [comment]
Well, there are I think *other* cases where it makes sense to be able to re-push an 'obsoleted' update. I think the same state is used for an update that has hit its negative karma threshold; the submitter should be able to edit the update (and fix the bug) and then resubmit it.

I think there may be scenarios where we might want to do the same with a truly 'obsoleted' update...like for instance if we pulled it into a candidate compose during a freeze and thus really need to push *that* update stable, not a later one, maybe?

**Comment by mattiaverga** [comment]
You're right, a negative karma threshold will cause the update to be obsoleted.
I think we should change this. I mean, what's the difference between "obsolete" and "unpush"? Looking in the code the `obsolete()` method removes ALL tags from the builds of the update, while `unpush()` moves the builds back into the `candidate` tag.

In my opinion we should "obsolete" an update when we're sure it will be no more useful (like in `obsolete_older_updates`), otherwise we should just "unpush" it (`obsolete_if_unstable()` should become `unpush_if_unstable()`).

I don't understand the other scenario you described.

**Comment by AdamWill** [comment]
It's specific to the release process. So, there's this thing we do where we pull builds from updates-testing into candidate composes. See e.g. the Beta release candidate compose request:

https://pagure.io/releng/issue/9319

the things listed under "Please include:" there are updates that were in updates-testing at the time we did the compose, but which we knew we needed to include in the compose in order to fix release-blocking or freeze exception issues. There just isn't always time to get those updates pushed stable before doing the compose, so releng has a procedure where they pull them into a side tag and include that side tag in the candidate compose.

However, we *do* want to make sure that the *exact* builds pulled into the candidate compose get pushed stable afterwards. Especially for Final - it's actually a requirement that we push the precise builds from the RC that is released as Final stable before we produce the final, frozen release trees and cut over to using the updates repo.

So, the problem is if this happens:

1. We have foo-1.0-1 in stable
2. Maintainer submits an update for foo-2.0-1 , fixing a release blocker bug
3. We pull that update into the release candidate compose
4. The maintainer submits an update for foo-3.0-1, obsoleting the foo-2.0-1 update, but this update doesn't fix any release blocker or FE bug
5. We approve the release candidate with foo-2.0-1 in it for release
6. We now need to push *exactly* the foo-2.0-1 update stable before finalizing the release tree. We cannot release with foo-1.0-1 in stable, or push foo-3.0-1 stable

**Comment by mattiaverga** [comment]
So, I think we can limit the power of resurrecting an obsoleted update only to provenpackagers?
I tried to summarize the lifetime of an Update in the following flowchart. While doing that I also discovered that it's possible to push back to testing/stable updates for an archived release... this is not so terrible, since the request will do nothing, but it would be nice to fix it.

![Untitled Diagram](https://user-images.githubusercontent.com/17218209/79734395-f0b23d00-82f6-11ea-9d04-ff5ffa922194.png)
